crates/bevy_pbr/src/light.rs

```diff
@@ -479,6 +590,98 @@ fn ndc_position_to_cluster(
         .clamp(UVec3::ZERO, cluster_dimensions - UVec3::ONE)
 }
 
+// Calculate an AABB for the light in view space, returns a (Vec3, Vec3) containing min and max with
+// - x and y in view space with range [-1, 1]
+// - z in world space with view orientation, with range [-inf, -0.0001] for perspective, and [1.0000, inf] for orthographic
+fn cluster_space_light_aabb(
+    is_orthographic: bool,
+    inverse_view_transform: Mat4,
+    projection_matrix: Mat4,
+    light_sphere: &Sphere,
+) -> (Vec3, Vec3) {
+    let light_aabb_view = Aabb {
+        center: (inverse_view_transform * light_sphere.center.extend(1.0)).xyz(),
+        half_extents: Vec3::splat(light_sphere.radius),
+    };
+    let (mut light_aabb_view_min, mut light_aabb_view_max) =
+        (light_aabb_view.min(), light_aabb_view.max());
+
+    if is_orthographic {
+        // constraint z to be positive - i.e. in front of the camera
+        light_aabb_view_min.z = light_aabb_view_min.z.max(1.0);
+        light_aabb_view_max.z = light_aabb_view_max.z.max(1.0);
+    } else {
+        // constraint z to be negative - i.e. in front of the camera
+        light_aabb_view_min.z = light_aabb_view_min.z.min(-0.0001);
+        light_aabb_view_max.z = light_aabb_view_max.z.min(-0.0001);
```

Comment:
I feel like this should be `.min(0.0)` and divide by zero cases should be handled elsewhere.

Reply:
Discussed on Discord. Trippy things happen in ndc when the view space light aabb extends behind the camera. In view space +x is to the right regardless of whether the position is in front of or behind the camera. But, in ndc, at view z = 0.0 ndc x,y is all undefined, and at view z > 0.0 (i.e. behind the camera) ndc +x is to the left and +y is down - they get flipped! This means that the spatial min/max that is going on gets broken. Projections...